PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,468 @@
+            <#
+                The * character specifies "0 or more" of the symbol or group that comes 
+                mmediately before it. In this case, the pattern can be read aloud as "zero or more
+                of the letter p, then the letters e a r s."
+            #>
+    
+            'pears' -match 'p*ears' | Should -BeTrue
+    
+            [bool]$starMatch = $____ # Should be either $true or $false
+            $starMatch | Should -Be 'shears' -match 'p*ears'
+    
+            <#
+                What happened in that last example? What part of 'shears' matched the pattern 
+                'p*ears'? When you use the -match operator, you can see some interesting information
+                in the automatic variable $matches. $matches is a collection of the match objects
+                that are created when you use the -match operator. This isn't as flexible nor as
+                robust as using [regex]::Matches(), but it's handy in a pinch.
+                
+                In this case, you'll see that the "sh" in "shears" weren't part of the match. The
+                pattern "zero or more p's, followed by e a r s" is found in the string, even though
+                there are parts of the string that aren't relevant to that matching effort.
+    
+                $matches[0] is always your most recent match.
+            #>
+    
+            '____' | Should -Be $matches[0]
+        }
+    }
+
+    Context '+' {
+        It 'specifies 1 or more of something' {
+            <#
+                The + symbol is a lot like the star, but instead of matching zero or more, it
+                matches one or more.
+            #>
+    
+            '____' -match 'p+ickles' | Should -BeTrue
+            '____' -match 'p+ickles' | Should -BeFalse
+        }
+    }
+
+    Context '?' {
+        It 'specifies 0 or 1 of something' {
+            # The ? matches "zero or one" of something.
+            
+            [bool]$boatMatch = $____ # Should be either $true or $false
+            $boatMatch | Should -Be 'flying through the sky' -match 'f+lying'
+            
+            [bool]$floatMatch = $____ # Should be either $true or $false
+            $floatMatch | Should -Be 'floating away' -match 'b+oat'
+        }
+    }
+
+    Context 'Greediness and Laziness' {
+        It 'matches a little or matches a lot' {
+            <#
+            A number of the quantifiers have two versions: A greedy version which tries to match an
+            element as many times as possible, and a non-greedy (or lazy) version which tries to
+            match an element as few times as possible. You can turn a greedy quantifier into a lazy
+            quantifier by simply adding a ?.
+            #>
+
+            $greedy = 'o+'
+            $lazy = 'o+?'
+            '____' | Should -Be @([regex]::Matches('helloooooo', $greedy).Value)[0]
+            '____' | Should -Be @([regex]::Matches('helloooooo', $lazy).Value)[0]
+        }
```

Comment:
This should be reflected with a more real initial example. For instance, you can put `$____` as the ::Matches() variable and have the user pick which is appropriate for two given initial strings and expected results.

E.g., with an initial string, you have something like `(aside) statement (another aside)`, one regex that matches `\(.+\)`, and one that matches `\(.+\)` (or something similar). You could do one where the student has to pick the right regex pattern to use, and one where they have to enter the right result strings. I know `.` isn't mentioned till a bit further down, so you can pick a different example to use if you'd like, but that showcases the idea I hope? Let me know if it's not clear! 💖 

Again as separate `It` blocks, so maybe one focused on lazy, one on greedy, and a last one where the student has to determine if the greedy or lazy approach is appropriate? 🤔

Reply:
I know that the example used isn't super relatable, but it's super simple, which I think is more important here.

I've added blocks that focus on lazy and greedy.